From fbff3cdb4598f39a6f73202bef6c30cd529af9d2 Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Tue, 16 Aug 2022 20:16:00 +0530
Subject: [PATCH] Disabling/Removing Unwanted Libraries & Services.

Removed Unwanted apps (for ex. NFC, Vibrator, GNSSS etc.)
as these are not required when running android on VM.
this is one combined patch having changes from below
owner as well-:
Tanuj Tekriwal tanuj.tekriwal@intel.com
Kaushlendra Kumar <kaushlendra.kumar@intel.com>

Tracked-On:
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 target/product/base_system.mk         |  4 ----
 target/product/default_art_config.mk  |  1 -
 target/product/handheld_product.mk    |  7 -------
 target/product/handheld_system.mk     | 29 ---------------------------
 target/product/handheld_system_ext.mk |  1 -
 target/product/handheld_vendor.mk     |  1 -
 6 files changed, 43 deletions(-)

diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index 14ce1af744..e2a9ba46db 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -239,7 +239,6 @@ PRODUCT_PACKAGES += \
     screencap \
     sdcard \
     secdiscard \
-    SecureElement \
     selinux_policy_system \
     sensorservice \
     service \
@@ -259,8 +258,6 @@ PRODUCT_PACKAGES += \
     svc \
     task_profiles.json \
     tc \
-    telecom \
-    telephony-common \
     tombstoned \
     traced \
     traced_probes \
@@ -362,7 +359,6 @@ PRODUCT_PACKAGES_DEBUG := \
     arping \
     dmuserd \
     gdbserver \
-    idlcli \
     init-debug.rc \
     iotop \
     iperf3 \
diff --git a/target/product/default_art_config.mk b/target/product/default_art_config.mk
index 0fa9058f0b..044f78f646 100644
--- a/target/product/default_art_config.mk
+++ b/target/product/default_art_config.mk
@@ -41,7 +41,6 @@ PRODUCT_BOOT_JARS += \
     framework-minus-apex \
     framework-graphics \
     ext \
-    telephony-common \
     voip-common \
     ims-common
 
diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
index 2199c570a0..c10bfe2e4f 100644
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -22,17 +22,10 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/media_product.mk)
 
 # /product packages
 PRODUCT_PACKAGES += \
-    Browser2 \
-    Calendar \
-    Camera2 \
-    Contacts \
-    DeskClock \
-    Gallery2 \
     LatinIME \
     Music \
     OneTimeInitializer \
     preinstalled-packages-platform-handheld-product.xml \
-    QuickSearchBox \
     SettingsIntelligence \
     frameworks-base-overlays
 
diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index b7a2d0dad0..23fe45a1b0 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -32,46 +32,17 @@ $(call inherit-product-if-exists, frameworks/base/data/keyboards/keyboards.mk)
 $(call inherit-product-if-exists, frameworks/webview/chromium/chromium.mk)
 
 PRODUCT_PACKAGES += \
-    BasicDreams \
-    BlockedNumberProvider \
-    Bluetooth \
-    BluetoothMidiService \
-    BookmarkProvider \
-    BuiltInPrintService \
-    CalendarProvider \
     cameraserver \
     CameraExtensionsProxy \
-    CaptivePortalLogin \
-    CertInstaller \
     clatd \
-    DocumentsUI \
-    DownloadProviderUi \
-    EasterEgg \
-    ExternalStorageProvider \
     FusedLocation \
     InputDevices \
-    KeyChain \
     librs_jni \
-    ManagedProvisioning \
-    MmsService \
     MtpService \
     MusicFX \
-    NfcNci \
-    PacProcessor \
-    PrintRecommendationService \
     PrintSpooler \
-    ProxyHandler \
     screenrecord \
-    SecureElement \
-    SharedStorageBackup \
-    SimAppDialog \
-    Telecom \
-    TelephonyProvider \
-    TeleService \
-    Traceur \
     UserDictionaryProvider \
-    VpnDialogs \
-    vr \
 
 
 PRODUCT_SYSTEM_SERVER_APPS += \
diff --git a/target/product/handheld_system_ext.mk b/target/product/handheld_system_ext.mk
index d935fbfddf..8406fee939 100644
--- a/target/product/handheld_system_ext.mk
+++ b/target/product/handheld_system_ext.mk
@@ -27,4 +27,3 @@ PRODUCT_PACKAGES += \
     Settings \
     StorageManager \
     SystemUI \
-    WallpaperCropper \
diff --git a/target/product/handheld_vendor.mk b/target/product/handheld_vendor.mk
index cb7cf741d4..75a0c41d39 100644
--- a/target/product/handheld_vendor.mk
+++ b/target/product/handheld_vendor.mk
@@ -25,5 +25,4 @@ PRODUCT_PACKAGES += \
     audio.primary.default \
     local_time.default \
     power.default \
-    vibrator.default \
 
-- 
2.17.1

